upup/pkg/fi/cloudup/vspheretasks/attachiso.go

```diff
@@ -17,19 +17,56 @@ limitations under the License.
 package vspheretasks
 
 import (
+	"bytes"
+	"fmt"
 	"github.com/golang/glog"
+	"github.com/pborman/uuid"
+	"io/ioutil"
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/pkg/model"
 	"k8s.io/kops/upup/pkg/fi"
 	"k8s.io/kops/upup/pkg/fi/cloudup/vsphere"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
 )
 
 // AttachISO represents the cloud-init ISO file attached to a VMware VM
 //go:generate fitask -type=AttachISO
 type AttachISO struct {
-	Name *string
-	VM   *VirtualMachine
```

Comment:
Hmm... I don't see this happening for slightly complex aws tasks: 
https://github.com/vmware/kops/blob/master/upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go
https://github.com/vmware/kops/blob/master/upup/pkg/fi/cloudup/awstasks/launchconfiguration.go